libs/agno/agno/vectordb/weaviate/weaviate.py

```diff
@@ -191,7 +311,17 @@ def upsert(self, documents: List[Document], filters: Optional[Dict[str, Any]] =
             filters (Optional[Dict[str, Any]]): Filters to apply while upserting
         """
         log_debug(f"Upserting {len(documents)} documents into Weaviate.")
-        self.insert(documents)
+
+    async def async_upsert(self, documents: List[Document], filters: Optional[Dict[str, Any]] = None) -> None:
+        """
+        Upsert documents into Weaviate asynchronously.
+
+        Args:
+            documents (List[Document]): List of documents to upsert
+            filters (Optional[Dict[str, Any]]): Filters to apply while upserting
+        """
+        log_debug(f"Upserting {len(documents)} documents into Weaviate asynchronously.")
```

Comment:
Weviate actually support replace as opposed to insert. So for both cases "upsert" is actaully possible

Reply:
done